docs/data/sql_functions.yml

```diff
@@ -516,7 +516,14 @@ conditional:
   - sql: NULLIF(value1, value2)
     description: Returns NULL if value1 is equal to value2; returns value1 otherwise. E.g., NULLIF(5, 5) returns NULL; NULLIF(5, 0) returns 5.
   - sql: COALESCE(value1, value2 [, value3]*)
-    description: Returns the first value that is not NULL from value1, value2, .... E.g., COALESCE(NULL, 5) returns 5.
+    description: |
+      Coalesce specifies a series of expressions, and returns the first expression whose value is not null.
+
+      If all the expressions evaluate as null, coalesce returns a null value.
+      The return type of the coalesce function is the least common type among its arguments, and it's nullable if all its arguments are nullable.
+
+      E.g. COALESCE(NULL, "-") returns "-".
+      E.g. COALESCE(f0, f1, "-") returns the first non-null value among f0 and f1, if both are null returns "-".
```

Comment:
Let's please turn this into a proper code block of examples. For example, look at the `JSON_*` functions we recently introduced. The examples should also be valid SQL, i.e. make use of SQL comments and proper syntax for string literals, something like

```
-- Returns 'default'
COALESCE(NULL, 'default')

-- Returns 'default' if f0 and f1 are both NULL
COALESCE(f0, f1, 'default')
```

Reply:
Also make sure to run the docs locally to verify everything's looking good. Unfortunately we currently need to manually make sure the code block does not scroll horizontally (which results in really poor UX), and line-break the examples if necessary.

Reply:
![Screenshot 2021-09-15 at 15-40-41 System (Built-in) Functions](https://user-images.githubusercontent.com/6706544/133444161-2564c852-afd9-454a-a438-c8cd2dbcfb1a.png)

Looks good to me